modules/nlu-testing/src/tools/data_loader.ts

```diff
@@ -0,0 +1,73 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+
+import { BotState, Data, RawData, Test } from '../backend/typings'
+
+export async function getTrainTestDatas(state: BotState, bp: typeof sdk) {
+  const EMBEDDING_FOLDER = './embeddings'
+  let vectorized_train: Data[] = []
+  let vectorized_test: Data[] = []
+
+  if (await state.ghost.fileExists(EMBEDDING_FOLDER, 'train_set.json')) {
+    vectorized_train = await state.ghost.readFileAsObject<Data[]>(EMBEDDING_FOLDER, 'train_set.json')
+  } else {
```

Comment:
this whole `else` block should be extracted in a private local function. Something like `createTrainEmbeddingsFile()`

Then your code could look like this:

```
const vectorsAlreadyExist = await state.ghost.fileExists(EMBEDDING_FOLDER, 'train_set.json')
const vectorized_train = vectorsAlreadyExist ? 
    await state.ghost.readFileAsObject<Data[]>(EMBEDDING_FOLDER, 'train_set.json') 
    : await createTrainEmbeddingsFile()
```